### PR TITLE
Upgrade Terminus version

### PR DESCRIPTION
**Pull request by srtfisher**
- Adds support for PHP 8.3 out of the box.
- Removes `env:deploy` call per https://github.com/pantheon-systems/terminus/commit/3faec26031ae0bc6286b1e2b0f1fb762956d98be / Terminus ^3.5.0.

### CODE REVIEW

**Comment by srtfisher** [comment]
Terminus now throws `This command should only be used to deploy to test or live environments` if called on a multidev.

File: action.yml [hunk 1]
```diff
@@ -64,16 +64,12 @@ runs:
       shell: bash
       run: terminus auth:login --machine-token=${{ inputs.pantheon_machine_token }}
     - name: "Deploy to Pantheon environment ${{ inputs.pantheon_env_name }}"
+      # Skipped if autopromote is false or we're not deploying to the dev environment.
+      if: ${{ inputs.autopromote == 'true' && (inputs.pantheon_env_name == 'production' || inputs.pantheon_env_name == 'master' || inputs.pantheon_env_name == 'dev') }}
```
Comment (benpbolton) [inline action.yml]: I believe this would prevent all deployments to multisite environments

**Review by benpbolton — CHANGES_REQUESTED** [review]
While this attempts a simplification, it makes the assumption that no deployments happen to `inputs.pantheon_env_name` other than `dev` ?

File: action.yml (hunk 1 above)
Comment (srtfisher) [reply action.yml:68]: It's deploying to a multidev right now for me.

Comment (srtfisher) [reply action.yml]: multidevs would have `autopromote` set to false.

**Comment by benpbolton** [comment]
CleanShot 2024-08-05 at 09 32 15

as of terminus 3.5,
> Env deploy command should only be used in test/live envs (#2575)

See also: https://github.com/pantheon-systems/terminus/pull/2575/files

**Review by benpbolton — APPROVED** [review]
Awesome, thanks

**Pull request merged**